src/scancode/resource.py

```diff
@@ -1444,11 +1445,10 @@ def _populate(self, scan_data):
             self.has_single_resource = True
         if not resources_data:
             raise Exception('Input has no file-level scan results.')
-        resources_data = iter(resources_data)
 
-        # The root MUST be the first resource. We use it as a template for
-        # actual Resource attributes
-        root_data = resources_data.next()
+        # We collect the first Resource so we can see what attributes it has and determine
```

Comment:
What is the empty `def _get_top_level_attributes(self, scan_data):` above?
Also IMHO we are starting to abuse inner nested functions in this function and this starts to make reading hard. It would be good to refactor that soon enough (and that's OK to do this with this PR)

Reply:
I left that method in on accident. I'll take a look at refactoring the code.

Reply:
I moved the in-line functions to be class methods and folded the logic of `get_resource_basic_attributes()` into the main Resource creation loop.